src/t_stream.c

```diff
@@ -1370,67 +1774,42 @@ int streamParseIntervalIDOrReply(client *c, robj *o, streamID *id, int *exclude,
     return C_OK;
 }
 
+/* We re-write trimming options that use 'approx'.
+ * otherwise trimming is no longer deterministic on replicas / AOF.
+ * Note: This function take ownership over 'arg' */
+void streamRewriteTrimArgument(client *c, robj* arg, int idx, int approx) {
+    if (approx) {
+        robj *equal_obj = createStringObject("=",1);
+        rewriteClientCommandArgument(c,idx-1,equal_obj);
+        decrRefCount(equal_obj);
+    }
+
+    rewriteClientCommandArgument(c,idx,arg);
+    decrRefCount(arg);
+}
+
 /* We propagate MAXLEN ~ <count> as MAXLEN = <resulting-len-of-stream>
- * otherwise trimming is no longer determinsitic on replicas / AOF. */
-void streamRewriteApproxMaxlen(client *c, stream *s, int maxlen_arg_idx) {
+ * otherwise trimming is no longer deterministic on replicas / AOF. */
+void streamRewriteMaxlen(client *c, stream *s, int maxlen_arg_idx, int approx) {
     robj *maxlen_obj = createStringObjectFromLongLong(s->length);
-    robj *equal_obj = createStringObject("=",1);
-
-    rewriteClientCommandArgument(c,maxlen_arg_idx,maxlen_obj);
-    rewriteClientCommandArgument(c,maxlen_arg_idx-1,equal_obj);
+    streamRewriteTrimArgument(c, maxlen_obj, maxlen_arg_idx, approx);
+}
 
-    decrRefCount(equal_obj);
-    decrRefCount(maxlen_obj);
+/* We propagate MINID ~ <id> as MINID = <resulting-first-id-of-stream>
+ * otherwise trimming is no longer deterministic on replicas / AOF. */
+void streamRewriteMinID(client *c, stream *s, int minid_arg_idx, int approx) {
+    streamID first_id;
+    streamGetEdgeID(s, 1, &first_id);
+    robj *minid_obj = createObjectFromStreamID(&first_id);
+    streamRewriteTrimArgument(c, minid_obj, minid_arg_idx, approx);
 }
 
-/* XADD key [MAXLEN [~|=] <count>] [NOMKSTREAM] <ID or *> [field value] [field value] ... */
+/* XADD key [MAXLEN [~|=] <count> | MINID [~|=] <id>| MAXAGE [~|=] <ms>] [NOMKSTREAM] <ID or *> [field value] [field value] ... */
```

Comment:
Also add MAXNODES? Even if it's undocumented, it should be documented here in the source code IMO.